src/main/java/UserAnalyzer.java

```diff
@@ -0,0 +1,20 @@
+/*
+This file contains UserAnalyzer class, which is the parent class for all analyzers
+ */
+package main.java;
+
+public class UserAnalyzer implements UserInformation{
+
+    /*
+    Below is the method that all analyzers share, the analyze method.
+    This method takes in any input of UserInformation, and stores it to the variable userInfo.
+    It then returns the type of output that will be specified in the other analyzers.
+    The exact contents will vary then.
+     */
+
+    public Object analyze(UserInformation userInfo){
```

Comment:
I assume that this method will be overridden in the future, but are you sure that "Object" would be the best signature for it? Will we change the method signature later once we start getting more specific?

Reply:
we were talking about how the return types of the subclasses will be different, and this is the solution we came up with for now. if you have a better suggestion that'd be appreciated

Reply:
If the return type is different maybe we can use Generics to restrict the return type? Just a thought.

Reply:
Generics might not really work because we can't really return back a type of UserAnalyzer? We might have to use Object for now.